circuit.py

```diff
@@ -104,46 +104,43 @@ def apply_transformation(self) -> None:
                 if isinstance(self.ops[index + 1], Measurement):
                     break
                 
-                self.commute_pi_over_four_rotation(index)
+                self.commute_rotation(index)
                 index += 1
 
 
-    def commute_pi_over_four_rotation(self, index: int) -> None:
+    def commute_rotation(self, index: int) -> None:
         """
-        Commute a pi/4 rotation block pass its' next block.
+        Commute a rotation block pass its neighbor block.
+
+        Args:
+            index (int): Index of the targeted block in the current circuit. 
+
         """
         next_block = index + 1
 
         if next_block >= len(self.ops):
             raise Exception("No operation to commute past")
 
-        if not isinstance(self.ops[index], Rotation) or not isinstance(self.ops[index+1], Rotation):
-            raise Exception("Can only commute rotations")
-
-        if not cast(Rotation,self.ops[index]).rotation_amount in {Fraction(1,4),Fraction(-1,4)}:
-            raise Exception("First operand must be +-pi/4 Pauli rotation")
-
```

Comment:
I would probably keep these two checks for robustness and self documentation, I know the caller never calls them on non pi/4 rotations but it's the kind of thing helps you catch bugs